config/grunt/custom-tasks/add-github-release.js

```diff
@@ -0,0 +1,139 @@
+const getUserInput = require( "../lib/get-user-input" );
+const githubApi = require( "../lib/github-api" );
+const uploadToGitHub = require( "../lib/upload-to-github" );
+
+/**
+ * Throws an error.
+ *
+ * @param {Object} response The response object.
+ * @param {Object} grunt    The grunt object.
+ *
+ * @returns {void}
+ */
+async function logError( response, grunt ) {
+	const responseObject = await response.json();
+
+	grunt.log.error( `Status: ${ response.status }` );
+	grunt.log.error( `Message: ${ responseObject.message }` );
+
+	if ( responseObject.errors ) {
+		responseObject.errors.forEach( error => {
+			grunt.log.error( JSON.stringify( error ) );
+		} );
+	}
+
+	grunt.fail.fatal( "Failed to create a pre-release on GitHub." );
+}
+
+/**
+ * Creates and pushes a GitHub (pre-)release and uploads the artifact to GitHub, using the GitHub API.
+ *
+ * @param {Object} grunt The grunt helper object.
+ * @returns {void}
+ */
+module.exports = function( grunt ) {
+	grunt.registerMultiTask(
+		"add-github-release",
+		"Creates and pushes a GitHub release and uploads the artifact to GitHub",
+		async function() {
+			const done = this.async();
+            let options = this.options({
+				manualEditChangelog: true,
+				target_commitish: '',
+				draft: false,
+				prerelease: true,
+				srcZipFilename: '',
+				dstZipFilename: '',
+				githubOwner:  '',
+				githubRepo: '',
+				enable: false,
+				uploadZip: false,
+			});
+			
+            
+			if ( ! options.enable) {
+				grunt.config.data.tagUrl =  'the github release was skipped'
+				return done();
+			}
+			
+			let changelog;
+			const owner = options.githubOwner;
+			const repo = options.githubRepo;
+
+			
+			if (options.manualEditChangelog === true) {
+				// Open a text editor to get the changelog.
+				changelog = await getUserInput( { initialContent: grunt.option( "changelog" ) } );
+			} else {
+				// use the changelog from the arguments
+				changelog = grunt.option( 'changelog' ) || process.env.RELEASECHANGELOG || grunt.config.data.changelog ;
+			}
+			// do some senity checking
+			if (changelog === '') {
+				grunt.fail.fatal( 'changelog entry required');
+			}
+			if (owner === '') {
+				grunt.fail.fatal( 'Github owner (organisation) required');
+			}
+			if (repo === '') {
+				grunt.fail.fatal( 'Github repo (repository) required');
+			}
+			if (options.srcZipFilename === '') {
+				grunt.fail.fatal( 'Github release source Zipfile name required');
+			}
+			if (options.dstZipFilename === '') {
+				grunt.fail.fatal( 'Github release destination Zipfile name required');
+			}
+			if (options.target_commitish === '') {
+				grunt.fail.fatal( 'Github release target_commitish name required');
+			}
+
+
+			const pluginVersion = grunt.file.readJSON( "package.json" ).yoast.pluginVersion;
+			grunt.verbose.writeln(pluginVersion);
+			/* eslint-disable camelcase */
+			const releaseData = {
+				tag_name: pluginVersion,
```

Comment:
Don't see a reason why this can't use camelCase formatting.

Reply:
the github variable set is called `target_commitish` I like to use the same variable names, unfortunately is github not using camelCase.

Reply:
in this case it even needs to be `target_commitish` as the `const response = await githubApi( path, releaseData, "POST");` needs this.